typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -319,7 +332,12 @@
                  "Type ~a could not be converted to a contract because it contains free variables."
                  type)))
             #`(#,(external-check-property #'#%expression check-valid-type)
-               #,(apply-contract ctc))])]))
+               #,(apply-contract
+                  (apply-contract
+                   #`(#,(casted-expr-property #'#%expression store-existing-type)
```

Comment:
Because that's where the `tc/#%expression` function looks for it, and that's what it already does for the `external-check-property` above it.